upf/upf_pfcp_api.c

```diff
@@ -803,9 +809,210 @@ lookup_nwi (u8 * name)
   return pool_elt_at_index (gtm->nwis, p[0]);
 }
 
+static bool
+validate_teid (u32 teid)
+{
+  return ((teid != 0) && (teid != 0xffffffff));
+}
+
+static u32
+generate_teid (u32 start, u32 mask)
+{
+  u8 ret_cnt = 0;
+  u32 c = 0;
+
+retry:
+
+  if (ret_cnt > 1)
+    return 0;
+
+  c = random_u32(&start) & (UINT32_MAX ^ mask);
+  if (!validate_teid(c))
+    {
+      //we are doing single try to regenerate, making next
+      //based on generated value
+      ret_cnt += 1;
+      goto retry;
+    }
+
+  return c;
+}
```

Comment:
this could also be moved into process_teid_generation

Reply:
Thanks Andreas. I will apply changes requested right after we will deal with PR#5 :)